test/common/json/gen_excluded_unicodes.cc

```diff
@@ -0,0 +1,127 @@
+#include "source/common/protobuf/utility.h"
+
+#include "test/common/json/utf8.h"
+
+#include "absl/strings/str_format.h"
+
+namespace Envoy {
+namespace Json {
+namespace {
+
+// Collects Unicode values that cannot be handled by the protobuf JSON encoder.
+// This is not needed for correct operation of the JSON sanitizer, but it is
+// needed for comparing sanitization results against the proto serializer, and
+// for differential fuzzing. We need to avoid comparing sanitization results for
+// strings containing utf-8 sequences that protobufs cannot serialize.
+//
+// bazel -c opt run test/common/json:gen_excluded_unicodes |& grep -v 'contains invalid UTF-8'
+//
+// The grep pipe is essential as otherwise you will be buried in thousands of
+// messages from the protobuf library that cannot otherwise be trapped. The
+// "-c opt" is recommended because otherwise it takes almost a minute to run
+// through all 4-byte UTF-8 sequences.
+//
+// Running in this mode causes two tests to fail, but prints two initialization
```

Comment:
Can we avoid the failure if possible? What I'm worried about is when the person upgrading protobuf library sees 3 tests fail or something and is uncertain on how to proceed. I think it's fine to have a rerun on each upgrade as needed, but it should be pretty bullet proof, right now it feels a bit wing-and-a-prayer.